### PR TITLE
update to use common source for operation includes

### PR DESCRIPTION
**Pull request by m-nash**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/Directory.Build.Common.targets [hunk 1]
```diff
@@ -188,7 +188,15 @@
   
   <!--TODO: end-->
 
-  <!-- *********** Management Client Library Override section ************* -->
+  <!-- *********** Files needed for LRO ************* -->
+  <ItemGroup Condition="'$(IncludeOperationsSharedSource)' == 'true'">
+    <Compile Include="$(AzureCoreSharedSources)OperationHelpers.cs" LinkBase="Shared/Core" />
```
Comment (Mohit-Chakraborty) [inline eng/Directory.Build.Common.targets:193]: Any particular reason we are moving the link from `Shared` to `Shared/Core`. Are we planning to have a hierarchy with different files at different levels.

**Review by Mohit-Chakraborty — APPROVED** [review]

File: eng/Directory.Build.Common.targets (hunk 1 above)
Comment (m-nash) [reply eng/Directory.Build.Common.targets:193]: The list of files for lro includes will be changing a few times as we do some refactoring so this makes it easier to make those changes as we move along

File: eng/Directory.Build.Common.targets [hunk 2]
```diff
@@ -188,7 +188,15 @@
   
   <!--TODO: end-->
 
-  <!-- *********** Management Client Library Override section ************* -->
+  <!-- *********** Files needed for LRO ************* -->
+  <ItemGroup Condition="'$(IncludeOperationsSharedSource)' == 'true'">
```
Comment (christothes) [inline eng/Directory.Build.Common.targets:192]: Do we need to define this here also?

https://github.com/Azure/azure-sdk-for-net/blob/50f7c35249fb263733d55edd1cf6662ede1fdbea/eng/Directory.Build.Common.props#L76

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Azure.AI.Language.Conversations.csproj [hunk 3]
```diff
@@ -5,6 +5,7 @@
     <Version>1.0.0-beta.3</Version>
     <PackageTags>Azure AI Language Conversations LUIS</PackageTags>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <IncludeOperationsSharedSource>true</IncludeOperationsSharedSource>
```
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Azure.AI.Language.Conversations.csproj:8]: Magic variables seem a lot harder to remember / learn. We already likely included a bunch of extra files like you're removing below so why make the process different? That just seems more complicated, especially for anyone onboarding to our code. Why can't new projects just do the same, or even a generator that might produce the .csproj add those 6 lines of XML to include the 4 files you need?

**Review by heaths — CHANGES_REQUESTED** [review]
Would love to discuss further the merits of including files directly as we already are likely doing vs. a new variable people have to know to use instead.

File: eng/Directory.Build.Common.targets (hunk 1 above)
Comment (christothes) [reply eng/Directory.Build.Common.targets:193]: I think @Mohit-Chakraborty 's question was why we didn't use `LinkBase="Shared"` like the other shared source files.

File: eng/Directory.Build.Common.targets (hunk 2 above)
Comment (heaths) [reply eng/Directory.Build.Common.targets:192]: Not all track 2 projects would need these files, I presume, such as those not based on HTTP e.g. Service Bus. Including them would unnecessary inflate package size.

Comment (m-nash) [reply eng/Directory.Build.Common.targets:192]: I don't think so because the idea is there are no default values you have to opt in

Comment (christothes) [reply eng/Directory.Build.Common.targets:192]: I meant define it and default it to false. If not defining it at all is equivalent, that's fine too.

File: eng/Directory.Build.Common.targets (hunk 1 above)
Comment (Mohit-Chakraborty) [reply eng/Directory.Build.Common.targets:193]: Correct. If we are going to have other subfolders as links, it makes sense.

**Review by heaths — APPROVED** [review]

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Azure.AI.Language.Conversations.csproj (hunk 3 above)
Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Azure.AI.Language.Conversations.csproj:8]: Per offline conversation, the idea was to make it easier to refactor LRO source changes without subsequent 50+ project updates every time.

Comment (m-nash) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Azure.AI.Language.Conversations.csproj:8]: The advantage here is that its one magic variable versus having to know a variable number of magic variables which are the list of files I need for LRO.  In this way you can simply say, give me the LRO files as one line and it doesn't matter what the list of files are.

Also we will be making changes to this list of files and being able to change it in one place is better than having to change 56 csproj files every time.

File: eng/Directory.Build.Common.targets (hunk 1 above)
Comment (heaths) [reply eng/Directory.Build.Common.targets:193]: Should all projects using only `LinkBase="Shared"` be updated to use `LinkBase="Shared/Core"` then so all files are visually co-located?

**Review by jsquire — COMMENTED** [review]
I wonder if it would make sense to consider creating an `Azure.Core.Shared` project explicitly for code sharing so that we could define sets of `projitems` to organize these kinds of bundles and make them more discoverable.

Examples:
- [Azure.Security.KeyVault.Shared.shproj](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Shared/src/Azure.Security.KeyVault.Shared.shproj)
- [Azure.Security.KeyVault.Shared.projitems](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Shared/src/Azure.Security.KeyVault.Shared.projitems)
- [Azure.Messaging.EventHubs.Shared.shproj](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Azure.Messaging.EventHubs.Shared.shproj)
- [Azure.Messaging.EventHubs.Shared.Authorization.projitems](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Azure.Messaging.EventHubs.Shared.Authorization.projitems)

@Mohit-Chakraborty may have some thoughts here...

**Comment by christothes** [comment]
> I wonder if it would make sense to consider creating an `Azure.Core.Shared` project explicitly for code sharing so that we could define sets of `projitems` to organize these kinds of bundles and make them more discoverable.
> 
> Examples:
> 
> * [Azure.Security.KeyVault.Shared.shproj](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Shared/src/Azure.Security.KeyVault.Shared.shproj)
> * [Azure.Security.KeyVault.Shared.projitems](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Shared/src/Azure.Security.KeyVault.Shared.projitems)
> * [Azure.Messaging.EventHubs.Shared.shproj](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Azure.Messaging.EventHubs.Shared.shproj)
> * [Azure.Messaging.EventHubs.Shared.Authorization.projitems](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Azure.Messaging.EventHubs.Shared.Authorization.projitems)
> 
> @Mohit-Chakraborty may have some thoughts here...

As I understand it, the Shared concept in those examples is to facilitate code to be shared across a multi-package client. In this case it is only a temporary (in theory) staging area for internal types to be used as shared source until the type becomes public in Azure.Core. In practice, we have lots of types that live here indefinitely, but that is not necessarily by design.

**Review by christothes — APPROVED** [review]

**Comment by jsquire** [comment]
> I believe the main reason for not making them public is that we aren't ready to commit them as part of the public API for some reason, including that future breaking changes may be possible or expected. In theory, types like this should not be long lived.

I don't believe that's the case for the majority of source shared by Core.   I'm thinking of many of the implementation types that aren't public-facing that are consumed by most of our SDKS.   For example:

- Argument
- AzureEventSource
- HashCodeBuilder
- AzureKeyCredentialPolicy


That said, I don't have a particularly strong opinion here.   I was just musing that a shared project may be helpful.

**Pull request merged**